ophyd/areadetector/base.py

```diff
@@ -134,9 +137,35 @@ def _array_shape_callback(self, **kwargs):
                        **self._metadata)
 
 
-class ADComponent(Component):
-    def __init__(self, cls, suffix=None, *, lazy=True, **kwargs):
-        super().__init__(cls, suffix=suffix, lazy=lazy, **kwargs)
+K = TypeVar("K", bound=OphydObject)
+
+
+class ADComponent(Component[K]):
```

Comment:
Is this representative of what downstream libraries with custom `Component` subclasses need to do to propagate the type checking niceties?

Reply:
Yes, my understanding is that this is required for subclasses of `Component`